extension/utils.js

```diff
@@ -51,3 +51,109 @@ function sanitizeDOM(dom) {
 	}
 	return dom;
 }
+
+async function postForm(form) {
+	// `content.fetch` is Firefox’s way to make fetches from the page instead of from a different context
+	// This will set the correct `origin` header without having to use XMLHttpRequest
+	// https://stackoverflow.com/questions/47356375/firefox-fetch-api-how-to-omit-the-origin-header-in-the-request
+	// https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_scripts#XHR_and_Fetch
+	const contentFetch = typeof window.content === 'object' ? window.content.fetch : window.fetch;
+	const formData = new FormData();
+	formData.append('utf8', '✓');
+	formData.append('authenticity_token', select('input[name="authenticity_token"]', form).value);
+
+	const response = await contentFetch(form.action, {
+		// TODO: drop `as` after https://github.com/microsoft/TSJS-lib-generator/issues/741
+		body: new URLSearchParams(formData),
+		method: 'POST',
+		headers: {
+			'Content-Type': 'application/x-www-form-urlencoded'
+		}
+	});
+
+	if (!response.ok) {
+		throw new Error(response.statusText);
+	}
+
+	return response;
+}
+
+const elements = new WeakMap();
+function _delegate(element, selector, type, callback, options) {
```

Comment:
It's best to keep all of `delegate-it` into its own `delegate-it.js` file

Same for `post-form.js`